File: javascript/leetcode/find-peak-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
    for (var i = 0; i < nums.length; i++) {
        if ((nums[i - 1] == undefined ? -Infinity : nums[i -1]) < nums[i] && nums[i] > (nums[i+1] == undefined ? -Infinity : nums[i+1])) {
            return i
        }
    }
    return false
};

File: ruby/leetcode/find-peak-element.rb
# @param {Integer[]} nums
# @return {Integer}
def find_peak_element(nums)
    for i in 0...nums.size() do
        if (i -1 < 0 || nums[i - 1] < nums[i]) && (i + 1 >= nums.size() || nums[i] > nums[i + 1])
            return i
        end      
    end
    return false
end
